src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Configuration/BaseProjectConfigurationDimensionProvider.cs

```diff
@@ -92,14 +102,13 @@ private async Task<ImmutableArray<string>> GetOrderedPropertyValuesAsync(Unconfi
             ImmutableArray<string> values = await GetOrderedPropertyValuesAsync(project);
             if (values.IsEmpty)
             {
-                return ImmutableArray<KeyValuePair<string, string>>.Empty;
+                return Enumerable.Empty<KeyValuePair<string, string>>();
             }
             else
             {
                 // First value is the default one.
-                var defaultValues = PooledArray<KeyValuePair<string, string>>.GetInstance();
-                defaultValues.Add(new KeyValuePair<string, string>(DimensionName, values[0]));
-                return defaultValues.ToImmutableAndFree();
+                KeyValuePair<string, string>[]? dimensionValues = new[] { new KeyValuePair<string, string>(DimensionName, values[0]) };
+                return dimensionValues.AsEnumerable();
```

Comment:
Can this not just be the following?

```suggestion
                return new[] { new KeyValuePair<string, string>(DimensionName, values[0]) };
```

I don't see what `AsEnumerable` does here other than allocate something we don't need. I might be missing something in the browser though.